### ISSUE TITLE
[amp-list-load-more] make "load-more-bookmark" bindable with infinite scroll

### ISSUE THREAD

**Issue by SethBuilder** [ampproject/amphtml#21434]
## Describe the new feature or change to an existing feature you'd like to see

[This ](https://envago.io)is my app.

I'm trying to re-implement my rather hacky list of travel adventures into an infinite scroll.

In my _new_ amp-list  (that is not online yet) I set this:
`src="/api/activity/all?country=&page=1"`
and this
`load-more-bookmark="next"`

Infinite scroll works fine so far. However, my situation is that the list updates its content based on two filters: _country_ and _adventure category_. Which means that after user interaction the `src` of the `amp-list` becomes:
  `[src]="myState.activities"`

This is one of the two ways of how I update the `[src]` based on country selection from a dropdown:

```
  <form method="GET"
  action="/api/activity/all?page=1"
  action-xhr="/api/activity/all?page=1"
  target="_top"
  id="countryForm"
  on="submit-success: AMP.pushState({
  myState: {
  activities: event.response.activities,
  categories: event.response.categories,
  next: event.response.next
}
});">
<input type="hidden"
name="country"
value=""
[value]="myState.selected ? myState.selected : ''">
<input type="hidden"
name="category"
value=""
[value]="myState.selectedCategory ? myState.selectedCategory : ''">
</form>
```

When the new list of adventure trips load (this time from `[src]`). Infinite scroll no longer works and only `?page=1` is requested.

This is the `amp-list` in full.
```
<div class="feed-container">
  <amp-list width="auto" height="1400"
  layout="fixed-height"
  src="/api/activity/all?country={{$userFromServedCountry ? $userFromServedCountry->code : 'me'}}&page=1"
  [src]="myState.activities"
  load-more="auto"
  binding="no"
  id="feed"
  load-more-bookmark="next"
  [load-more-bookmark]="myState.next"
  items="activities">
  <template type="amp-mustache">
    <div class="pin-container">
      <div style="position: relative;height: 100%" class="img-container">
        <amp-img class="cover" src="@{{inspiration.desktopInspiration}}" layout="fill"></amp-img>
      </div>
      <div class="details">
        <div>
          <div class="trip-name">
            <amp-img src="@{{category.hoverIcon}}" width="25" height="25" layout="fixed"></amp-img>
            <h4>@{{name}}</h4>
          </div>
          <div class="trip-name">
            <amp-img src="https://envagoprod.imgix.net/pint.svg?auto=compress,format" width="20" height="20" layout="fixed"></amp-img>
            <h4>@{{locationDisplayName}}</h4>
          </div>
        </div>
        <div class="price-container">
          <div class="price">
            <p><span class="book">Book</span> | @{{budget}} @{{currency}}</p>
          </div>
        </div>
      </div>
    </div>
  </template>

  <amp-list-load-more load-more-button>
    <button>See More</button> /* My custom see more button */
  </amp-list-load-more>

  <amp-list-load-more load-more-failed>
    <button>Unable to Load More</button>
  </amp-list-load-more>

  <amp-list-load-more load-more-end>
    Congratulations! You've reached the end. /* Custom load-end element */
  </amp-list-load-more>
</amp-list>
</div>
```

## Suggested feature for this issue:
to be able to bind `next `like this:
`[load-more-bookmark]="myState.next"`

Currently , doing so generate this error:

> amp-list#feed.i-amphtml-element.i-amphtml-layout-fixed-height.i-amphtml-layout-size-defined.i-amphtml-layout.i-amphtml-error.i-amphtml-element-error: Binding to [%s] on <%s> is not allowed. amp-bind load-more-bookmark AMP-LIST​​​

## Describe alternatives you've considered

My current pagination solution that is currently [online ](https://envago.io)is inspired by [AmpbyExample's example](https://ampbyexample.com/advanced/show_more_button/).

## Additional context

![image](https://user-images.githubusercontent.com/4164001/54439733-10655200-4742-11e9-8266-9e0c7a841143.png)

**Comment by cathyxz** [comment]
Thanks for trying out infinite scroll and taking the time to provide feedback. I read your comment a few times and I think maybe what you're trying to do is possible with the current feature set, so long as the bound `[src]` is a URL and not an object in `<amp-state>`. 

A few clarification questions: 
1. What does the sample json for your load-more endpoint look like? 
2. What are the post-bind values for `activities: event.response.activities` expected to look like? 
3. What does sample json from that endpoint look like? If it is returning a correct "next" value, but infinite scroll isn't working, then that is probably a bug. 

Here's a working example with code that tries to do something similar, but the backend is responsible for specifying the correct `next` url in JSON:  https://amp-infinite-scroll.glitch.me/bind.html. 

The relevant portions of sample code are as follows: 
```
  <amp-state id="data">
    <script type="application/json">
      {
        "items": 8,
        "left": 5, 
        "latency": 0, 
        "url": "//amp-infinite-scroll.glitch.me/generate?items=8&left=2"
      }
    </script>
  </amp-state>
  
<article>
  <h1>AMP List</h1>
  <p>This is a flexbox amp-list with tiles that allows you to adjust values for the backend endpoint.</p>
  <div>
    <span>Items per page: </span>
    <input type="text" id="items" placeholder="Items per page" value="8"
         on="change:AMP.setState({data: {items: event.value}})">
  </div>
  
  <div>
    <span>Number of additional pages: </span>
    <input type="text" id="left" placeholder="Pages left" value="2" 
         on="change:AMP.setState({data: {left: event.value}})">
  </div>
  
  <div>
    <span>Load latency: </span>
    <input type="text" id="latency" placeholder="Latency" value="0"
         on="change:AMP.setState({data: {latency: event.value}})">
  </div>


  <button
          on="tap:list.changeToLayoutContainer(),
              AMP.setState({data: {url: '//amp-infinite-scroll.glitch.me/generate?items=' 
                + data.items + '&left=' + data.left + '&latency=' + data.latency}})">
    Update
  </button>
  <div>
    <amp-list id="list" width="auto" height="200"
      layout="fixed-height"
      src="//amp-infinite-scroll.glitch.me/generate?items=8&left=5"
      [src]="data.url"
      load-more="auto"
      reset-on-refresh
      load-more-bookmark="next">
      <template type="amp-mustache">
        <div class="pin-container">
          <amp-img src="{{imageUrl}}" width="160" height="160"></amp-img>
          <div class="title">{{title}}</div>
          <div class="description">
            At vero eos et accusamus et iusto odio dignissimos ducimus...
          </div>
        </div>
      </template>
      <amp-list-load-more load-more-end>ENDED</amp-list-load-more>
    </amp-list>
  </div>
```

**Comment by cathyxz** [comment]
Hi @SethBuilder, let me know if you have further questions, or if the above sample isn't what you're looking for.

**Comment by cathyxz** [comment]
Closing due to inactivity.
